trollsched/shapefiles_from_schedule.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024 Pytroll community
+
+# Author(s):
+
+#   Adam Dybbroe <Firstname.Lastname @ smhi.se>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Create shapefiles of satellite instrument swath outlines from an XML pass reception plan."""
+
+
+import argparse
+import logging
+from datetime import datetime
+from pathlib import Path
+
+import defusedxml.ElementTree as ET
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import pyproj
+from shapely.geometry import Polygon
+from shapely.ops import transform
+
+from trollsched import INSTRUMENT, SATELLITE_NAMES
+from trollsched.logger import setup_logging
+from trollsched.satpass import create_pass
+
+logger = logging.getLogger(__name__)
+
+
+def get_shapely_polygon_from_lonlat(lons, lats):
+    """From a arrays of lons and lats return a shapely Polygon."""
+    geodata = np.vstack((lons, lats)).T
+    return Polygon(np.rad2deg(geodata))
+
+def create_shapefile_filename(satellite_pass_obj):
+    """From a trollsched.satpass instance create the shapefile filename."""
+    satname = satellite_pass_obj.satellite.name.replace(" ", "-")
+    prefix = f"{satellite_pass_obj.instrument}_{satname}"
+    return f"{prefix}_{satellite_pass_obj.risetime:%Y%m%d%H%M}_{satellite_pass_obj.falltime:%Y%m%d%H%M}_outline.shp"
+
+def create_shapefile_from_pass(sat_pass, output_filepath):
+    """From a satellite overpass (instrument scanning outline) create a shapefile and save."""
+    sat_poly = get_shapely_polygon_from_lonlat(sat_pass.boundary.contour_poly.lon,
+                                               sat_pass.boundary.contour_poly.lat)
+
+    wgs84 = pyproj.CRS("EPSG:4326")  # WGS 84
+    a_crs = {"init": "epsg:4326"}
+
+    project = pyproj.Transformer.from_crs(wgs84, a_crs, always_xy=True).transform
+    new_shapes = transform(project, sat_poly)
+
+    gpd.GeoDataFrame(pd.DataFrame(["p1"], columns=["geom"]), crs=a_crs,
+                     geometry=[new_shapes]).to_file(output_filepath)
+
+
+def shapefiles_from_schedule_xml_requests(filename, satellites, tle_file, output_dir):
```

Comment:
Would it be very complicated to add a test for this?

Reply:
Added a simple test, thanks!